typescript/sdk/src/providers/SmartProvider/SmartProvider.foundry-test.ts

```diff
@@ -142,4 +142,38 @@ describe('SmartProvider', async () => {
 
     expect(erc20.address).to.not.be.empty;
   });
+
+  it('returns the blockchain error reason: "ERC20: transfer to zero address"', async () => {
+    const smartProvider = HyperlaneSmartProvider.fromRpcUrl(NETWORK, URL, {
+      maxRetries: 1,
+    });
+    const signer = new Wallet(PK, smartProvider);
+
+    const factory = new ERC20__factory(signer);
+    const token = await factory.deploy('fake', 'FAKE');
+    try {
+      await token.transfer(constants.AddressZero, 1000000);
+    } catch (e: any) {
+      expect(e.error.message).to.equal(
+        'execution reverted: revert: ERC20: transfer to the zero address',
+      );
+    }
+  });
+
+  it('returns the blockchain error reason: "ERC20: transfer amount exceeds balance"', async () => {
+    const smartProvider = HyperlaneSmartProvider.fromRpcUrl(NETWORK, URL, {
+      maxRetries: 1,
+    });
+    const signer = new Wallet(PK, smartProvider);
+
+    const factory = new ERC20__factory(signer);
+    const token = await factory.deploy('fake', 'FAKE');
+    try {
+      await token.transfer(signer.address, 1000000);
+    } catch (e: any) {
+      expect(e.error.message).to.equal(
```

Comment:
Hey @jmrossy 
I want to point out that when I throw the blockchain errors [here](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4258/files#diff-cb1739faa2e4859a713dabebcc65571ea40d828b5d32bff34327ece27b7ca973R446), and then catch it in this test, it becomes "wrapped". This is why i need to do `e.error.message`. Any ideas why?

Heres the console.log:
```
Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (error={}, tx={"data":"0xa9059cbb000000000000000000000000f39fd6e51aad88f6f4ce6ab8827279cfffb9226600000000000000000000000000000000000000000000000000000000000f4240","to":{},"from":"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266","type":2,"maxFeePerGas":{"type":"BigNumber","hex":"0x82f1f5e0"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x3b9aca00"},"nonce":{},"gasLimit":{},"chainId":{}}, code=UNPREDICTABLE_GAS_LIMIT, version=abstract-signer/5.7.0)
    at Logger.Logger.makeError (/Users/leyu/Desktop/Code/hyperlane/hyperlane-monorepo/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
    at Logger.Logger.throwError (/Users/leyu/Desktop/Code/hyperlane/hyperlane-monorepo/node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
    at <anonymous> (/Users/leyu/Desktop/Code/hyperlane/hyperlane-monorepo/node_modules/@ethersproject/abstract-signer/src.ts/index.ts:301:31)
    at async Promise.all (index 7) {
  reason: 'cannot estimate gas; transaction may fail or may require manual gas limit',
  code: 'UNPREDICTABLE_GAS_LIMIT',
  error: Error: execution reverted: revert: ERC20: transfer amount exceeds balance
      at HyperlaneSmartProvider.throwCombinedProviderErrors (/Users/leyu/Desktop/Code/hyperlane/hyperlane-monorepo/typescript/sdk/src/providers/SmartProvider/SmartProvider.ts:452:13)
      at HyperlaneSmartProvider.performWithFallback (/Users/leyu/Desktop/Code/hyperlane/hyperlane-monorepo/typescript/sdk/src/providers/SmartProvider/SmartProvider.ts:357:14)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at retryAsync (/Users/leyu/Desktop/Code/hyperlane/hyperlane-monorepo/typescript/utils/src/async.ts:71:22),
  tx: {
    data: '0xa9059cbb000000000000000000000000f39fd6e51aad88f6f4ce6ab8827279cfffb9226600000000000000000000000000000000000000000000000000000000000f4240',
    to: Promise { '0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9' },
    from: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
    type: 2,
    maxFeePerGas: BigNumber { _hex: '0x82f1f5e0', _isBigNumber: true },
    maxPriorityFeePerGas: BigNumber { _hex: '0x3b9aca00', _isBigNumber: true },
    nonce: Promise { 5 },
    gasLimit: Promise { <rejected> [Circular *1] },
    chainId: Promise { 31337 }
  }
}
```

As you can see, the thrown error from the `throwCombinedProviderErrors()` is inside this caught error's `error` property. It's surprising because I thought it would just be a single error, and not a nested one.

Reply:
Sorry but I'm not sure. Maybe the test framework is catching and wrapping it or maybe there's some of our own SDK code running between here and there that catches + rethrows